MetaMorpheus/GUI/Views/CustomProteaseWindow.xaml.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Proteomics.ProteolyticDigestion;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.IO;
+using EngineLayer;
+using System.Xml;
+using Easy.Common.Extensions;
+using System.Linq;
+using MassSpectrometry;
+using Omics.Digestion;
+
+namespace MetaMorpheusGUI.Views
+{
+    /// <summary>
+    /// Interaction logic for CustomProteaseWindow.xaml
+    /// </summary>
+    public partial class CustomProteaseWindow : Window
+    {
+        public CustomProteaseWindow()
+        {
+            InitializeComponent();
+
+            foreach (CleavageSpecificity c in Enum.GetValues(typeof(CleavageSpecificity)))
+            {
+                CleavageSpecComboBox.Items.Add(c);
+            }
+        }
+
+        private void SaveProteaseButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (ProteaseNameTextBox.Text.Length == 0)
+            {
+                MessageBox.Show("Please specify the name of the protease");
+                return;
+            }
+            if (CleavageSpecComboBox.Text.Length == 0)
+            {
+                MessageBox.Show("Please specify the cleavage specificity of the protease");
+                return;
+            }
+            if (DigestionMotifTextBox.Text.Length == 0)
+            {
+                MessageBox.Show("Please specify the digestion motif(s) of the protease");
+                return;
+            }
+            string proteaseName = ProteaseNameTextBox.Text;
+            string psiMsAccessionNum = PsiMsAccessionNumTextBox.Text;
+            string psiMsName = PsiNameTextBox.Text;
+            string motifString = DigestionMotifTextBox.Text;
+            Dictionary<string, CleavageSpecificity> CleavageSpecDic = new Dictionary<string, CleavageSpecificity>();
+            foreach (CleavageSpecificity c in Enum.GetValues(typeof(CleavageSpecificity))) {
+                CleavageSpecDic.Add(c.ToString(), c);
+            }
+            CleavageSpecificity cleavageSpecificity = CleavageSpecDic[CleavageSpecComboBox.Text];
+            List<DigestionMotif> digestionMotifs;
```

Comment:
Replace this dictionary creation with 
if (Enum.TryParse(CleavageSpecComboBox.Text, out _ cleaveageSpec) 
{
     set the value
}
else
{
   message box error and return
}